cmake/mongocrypt-config.cmake

```diff
@@ -40,3 +40,24 @@ if (_using_system_intel_dfp)
       PROPERTY IMPORTED_LOCATION "${_MONGOCRYPT_SYSTEM_INTEL_DFP_STATIC}"
       )
 endif ()
+
+find_dependency(Threads)
+
+# Link for dlopen():
+set_property (TARGET mongo::mongocrypt::platform APPEND PROPERTY INTERFACE_LINK_LIBRARIES ${CMAKE_DL_LIBS})
```

Comment:
Is the `mongo::mongocrypt::platform` target expected to be used outside of `libmongocrypt`?

Reply:
Not directly, but it will appear as a transitive usage requirement on libmongocrypt consumers.